src/models/controllercapabilities.h

```diff
@@ -0,0 +1,64 @@
+#ifndef CONTROLLERCAPABILITIES_H
+#  define CONTROLLERCAPABILITIES_H
+
+#  include <QObject>
+#  include <QString>
+
+/*
+ * This Model defines which capabilitie a controller has.
+ */
+class ControllerCapabilities : public QObject {
+  Q_OBJECT
+
+ public:
+  // Serialises the Obj
+  QJsonObject serialize();
+  // Takes json and populates properties
+  void read(const QJsonObject& obj);
+
+  ControllerCapabilities() {}
+  ControllerCapabilities(const ControllerCapabilities& other);
+  ControllerCapabilities& operator=(const ControllerCapabilities& other);
+
+  // Each capability is defined by it's supportlevel
+  enum SupportLevel {
+    // Feature is supported and ready to use
+    Supported = 1,
+    // Not known yet, might happen if the supportlevel is checked
+    // before the controller is initialised
+    Unknown = 0,
+    // Feature is supported but currently unavailable
+    // e.g: "failed to communicate with the windows-split tunnel driver"
+    Deactivated = -1,
+    // Feature is not supported by Controller / OS
+    Unsupported = -2,
+  };
+  Q_ENUM(SupportLevel);
+
+#  define Capability(name, defaultValue) \
```

Comment:
As a preference thing, I am of the opinion that all macros should be uppercase unless they are specifically trying to emulate a standard function that isn't uppercase.